### PR TITLE
made v-show respect display values in style attributes

### PR DESCRIPTION
**Pull request by chrisvfritz**
Addresses issue described [here](https://github.com/vuejs/vue/issues/2873#issuecomment-237921731).

### CODE REVIEW

**Comment by miljan-aleksic** [comment]
Thank you @chrisvfritz, I hope the PR will be accepted :)

File: src/platforms/web/runtime/directives/show.js [hunk 1]
```diff
@@ -17,22 +17,24 @@ export default {
     if (value && transition && transition.appear && !isIE9) {
       enter(vnode)
     }
-    el.style.display = value ? '' : 'none'
+    const originalDisplay = el.style.display
+    el.style.display = value ? originalDisplay : 'none'
+    el.dataset.__vOriginalDisplay = originalDisplay
```
Comment (yyx990803) [inline src/platforms/web/runtime/directives/show.js]: I don't think IE9~10 supports `dataset`, so we may have to store the value directly on the element itself.

Comment (chrisvfritz) [inline src/platforms/web/runtime/directives/show.js]: @yyx990803 Wow, you're absolutely right - I can't believe I didn't know this! How should we handle this with the Flow types? Declare a custom type that extends `HTMLElement` and allows `__vOriginalDisplay` as a string?

File: src/platforms/web/runtime/directives/show.js [hunk 2]
```diff
@@ -17,7 +17,9 @@ export default {
     if (value && transition && transition.appear && !isIE9) {
       enter(vnode)
     }
-    el.style.display = value ? '' : 'none'
+    const originalDisplay = el.style.display
+    el.style.display = value ? originalDisplay : 'none'
+    el.__vOriginalDisplay = originalDisplay
```
Comment (yyx990803) [inline src/platforms/web/runtime/directives/show.js]: flow doesn't allow attaching properties to HTMLElements, let's just declare `el`'s type as `any` in the function signatures to make it pass.

**Pull request merged**